pkg/metrics/metrics.go

```diff
@@ -0,0 +1,188 @@
+package metrics
+
+import (
+	"encoding/json"
+	"strings"
+
+	corelisters "k8s.io/client-go/listers/core/v1"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	configv1 "github.com/openshift/cluster-samples-operator/pkg/apis/samples/v1"
+	"github.com/openshift/cluster-samples-operator/pkg/client"
+	sampoplisters "github.com/openshift/cluster-samples-operator/pkg/generated/listers/samples/v1"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	separator       = "_"
+	sampleNamespace = "openshift" + separator + "samples"
+	// per https://prometheus.io/docs/practices/naming/#metric-names feel our metrics here fall under the _info category of classification
+	info            = "info"
+	statusSubsystem = "status"
+
+	// openshift_samples_failed_imagestream_import_info
+	// since we want to index by imagestream name, where each imagestream is either failing or not failing, this is why
+	// we went with the _info suffix vs. the _total suffix
+	failedImportsQuery = sampleNamespace + separator + "failed" + separator + "imagestream" + separator + "import" + separator + info
+
+	degraded = "degraded" + separator + info
+	// openshift_samples_status_degraded_info
```

Comment:
if you already write it out here, why not just always write out the string? fwiw that's the most common practice as it's easy to search for metrics that way

Reply:
yeah there is some history behind all this wrt unfamiliarity with naming conventions, with some programming style opinions mixed into it ... even all the way back to when we first introduced metrics into openshift builds in 3.10

I've never felt strongly either way, and at least this morning, the idea of some simplification appeals to me :-)

will update in next push - thanks